### PR TITLE
Users/ajya/variable substitution fix master

### PR DESCRIPTION
**Pull request by Ajay-MS**

### CODE REVIEW

File: Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ps1 [hunk 1]
```diff
@@ -100,6 +115,13 @@ try{
 	Write-Verbose "Completed AzureRM WebApp Deployment Task"
 
 } finally {
+    if( $VariableSubstitution -eq "true")
+    {
+        if( Test-Path $packageFilePath )
+        {
+            Remove-Item -Force $packageFilePath
```
Comment (kmkumaran) [inline Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ps1]: I believe you wanted to use rmdir as Remove-Item was causing path limit exceeding error.

File: Tasks/AzureRmWebAppDeployment/task.json [hunk 2]
```diff
@@ -132,6 +137,15 @@
             "helpMarkDown": "Additional Web Deploy arguments that will be applied when deploying the Azure Web App like,-disableLink:AppPoolExtension -disableLink:ContentExtension."
         },
         {
+            "name": "VariableSubstitution",
+            "type": "boolean",
+            "label": "Variable Substitution",
+            "required": false,
+            "defaultValue": "false",
+            "groupName": "configtransforms",
+            "helpMarkDown": "Variables defined in the Defination will be automatically substitued in the appSettings,applicationSettings and connectionStrings section of the config files. If the same variable are defined in the Defination and Environment, then the En variables will be supersede the defination variables. Variable substitution is run after config Transformations."
```
Comment (kmkumaran) [inline Tasks/AzureRmWebAppDeployment/task.json]: We will have to add that secure variable support is not available for substitution.

File: Tasks/AzureRmWebAppDeployment/dependencies.json [hunk 3]
```diff
@@ -10,6 +10,16 @@
               "MSDeploy3.6/Microsoft.Web.Deployment.Tracing.dll",
               "MSDeploy3.6/msdeploy.exe"
             ]
+        },
+		    {
+            "archiveName": "7zip.zip",
+            "url": "https://vstsagenttools.blob.core.windows.net/tools/7zip/1/7zip.zip",
```
Comment (kmkumaran) [inline Tasks/AzureRmWebAppDeployment/dependencies.json]: We need to add appropriate license and EULA files for this. Can you check with @RoopeshNair for this.

File: Tasks/AzureRmWebAppDeployment/CompressionUtility.ps1 [hunk 4]
```diff
@@ -0,0 +1,48 @@
+﻿function UnzipWebDeployPkg
```
Comment (ericsciple) [inline Tasks/AzureRmWebAppDeployment/CompressionUtility.ps1]: You can simply use the .Net classes to do the unzip.

.Net 4.5 is already a demand of the 1.x Windows agent. We are adding prereq validation in the 2.x Windows agent.

Comment (Ajay-MS) [inline Tasks/AzureRmWebAppDeployment/CompressionUtility.ps1]: @ericsciple We are planning to support other compression formats in future e.g. war , tar.gz . That's why we are taking dependency on 7zip.

**Pull request closed without merging**